### PR TITLE
[Clang-Tidy] readability-inconsistent-declaration-parameter-name

### PR DESCRIPTION
**Pull request by veblush**
https://clang.llvm.org/extra/clang-tidy/checks/readability-inconsistent-declaration-parameter-name.html

Fix them manually.

### CODE REVIEW

File: src/core/lib/surface/call.cc [hunk 1]
```diff
@@ -637,11 +637,10 @@ static void execute_batch_in_call_combiner(void* arg, grpc_error* /*ignored*/) {
 
 // start_batch_closure points to a caller-allocated closure to be used
 // for entering the call combiner.
-static void execute_batch(grpc_call* call,
-                          grpc_transport_stream_op_batch* batch,
+static void execute_batch(grpc_call* call, grpc_transport_stream_op_batch* op,
```
Comment (markdroth) [inline src/core/lib/surface/call.cc:640]: The parameter here should be called `batch`, not `op`.  Let's fix the declaration instead.

File: src/core/lib/transport/transport.cc [hunk 2]
```diff
@@ -177,33 +177,31 @@ grpc_endpoint* grpc_transport_get_endpoint(grpc_transport* transport) {
 // though it lives in lib, it handles transport stream ops sure
 // it's grpc_transport_stream_op_batch_finish_with_failure
 void grpc_transport_stream_op_batch_finish_with_failure(
-    grpc_transport_stream_op_batch* batch, grpc_error* error,
+    grpc_transport_stream_op_batch* op, grpc_error* error,
```
Comment (markdroth) [inline src/core/lib/transport/transport.cc:180]: The parameter here should be called `batch`, not `op`.  Let's fix the declaration instead.

File: src/core/lib/transport/transport.cc [hunk 3]
```diff
@@ -226,11 +224,11 @@ static void destroy_made_transport_op(void* arg, grpc_error* error) {
   delete op;
 }
 
-grpc_transport_op* grpc_make_transport_op(grpc_closure* on_complete) {
+grpc_transport_op* grpc_make_transport_op(grpc_closure* on_consumed) {
```
Comment (markdroth) [inline src/core/lib/transport/transport.cc:227]: This should be called `on_complete`.  Let's fix the declaration instead.

File: src/core/lib/transport/transport.cc [hunk 4]
```diff
@@ -249,13 +247,13 @@ static void destroy_made_transport_stream_op(void* arg, grpc_error* error) {
 }
 
 grpc_transport_stream_op_batch* grpc_make_transport_stream_op(
-    grpc_closure* on_complete) {
+    grpc_closure* on_consumed) {
```
Comment (markdroth) [inline src/core/lib/transport/transport.cc:250]: Same here.

File: src/core/tsi/alts/handshaker/alts_tsi_handshaker.h [hunk 5]
```diff
@@ -78,20 +78,20 @@ tsi_result alts_tsi_handshaker_create(
  * - resp: data received from the handshaker service.
  * - is_client: a boolean value indicating if the result belongs to a
  *   client or not.
- * - result: address of ALTS TSI handshaker result instance.
+ * - self: address of ALTS TSI handshaker result instance.
  */
 tsi_result alts_tsi_handshaker_result_create(grpc_gcp_HandshakerResp* resp,
                                              bool is_client,
-                                             tsi_handshaker_result** result);
+                                             tsi_handshaker_result** self);
```
Comment (markdroth) [inline src/core/tsi/alts/handshaker/alts_tsi_handshaker.h:85]: This should be called  `result`.  Let's fix the impl instead.

File: src/core/tsi/alts/handshaker/alts_tsi_handshaker.h [hunk 6]
```diff
@@ -78,20 +78,20 @@ tsi_result alts_tsi_handshaker_create(
...
 /**
  * This method sets unused bytes of ALTS TSI handshaker result instance.
  *
- * - result: an ALTS TSI handshaker result instance.
+ * - self: an ALTS TSI handshaker result instance.
  * - recv_bytes: data received from the handshaker service.
  * - bytes_consumed: size of data consumed by the handshaker service.
  */
-void alts_tsi_handshaker_result_set_unused_bytes(tsi_handshaker_result* result,
+void alts_tsi_handshaker_result_set_unused_bytes(tsi_handshaker_result* self,
```
Comment (markdroth) [inline src/core/tsi/alts/handshaker/alts_tsi_handshaker.h:94]: Same here.

File: src/core/tsi/ssl_transport_security.h [hunk 7]
```diff
@@ -199,7 +199,7 @@ tsi_result tsi_ssl_client_handshaker_factory_create_handshaker(
 /* Decrements reference count of the handshaker factory. Handshaker factory will
  * be destroyed once no references exist. */
 void tsi_ssl_client_handshaker_factory_unref(
-    tsi_ssl_client_handshaker_factory* factory);
+    tsi_ssl_client_handshaker_factory* self);
```
Comment (markdroth) [inline src/core/tsi/ssl_transport_security.h:202]: This should be called `factory`.

**Review by markdroth — APPROVED** [review]
All comments are minor; feel free to merge after addressing.  Thanks!

**Comment by veblush** [comment]
Thank you for the review, Mark!

**Pull request merged**